lte/gateway/c/core/oai/include/proto_map.hpp

```diff
@@ -170,15 +170,15 @@ struct proto_map_s {
   **              the map. If key does not exists returns error             **
   **                                                                        **
   ***************************************************************************/
-  proto_map_rc_t remove(const keyT key) {
+  proto_map_rc_t remove(const keyT key, bool free_an_entry = true) {
```

Comment:
What is the need of this flag? Shouldn`t the entry be freed always?

Reply:
If flag is set, an entry for corresponding key would be removed and also contents are freed.
But in this PR, sgw_context would be initially in map, temporary_create_session_procedure_id_map. On reception of create session response, sgw_context would be removed from temporary_create_session_procedure_id_map and inserted into state_tied_map.